cmd/snap-recovery-chooser/menu.go

```diff
@@ -0,0 +1,66 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package main
+
+import (
+	"strings"
+)
+
+// Menu describes a menu with multiple entries
+type Menu struct {
```

Comment:
We need to discuss this, it feels premature given that we still need to discuss UI and some details might make this too simple, also while it covers anything that can do up/down/ok/back with a display or similar, there are devices that probably can do only less than that, there are also display size/translation issues.

My own idea so far was to have JSON like:

```
{
   "recovery-systems": [{
       "label": ...
       "model":  ...
       "model-display-name": ...
       "brand": ....
       "actions": [  // supported actions for this recovery system
          ...
       ],
       // other metadata
   },
   ...]
}
```

Purely in theory we could do both with a JSON object with both the high-level information on the recovery systems, and a more concrete menu structure as well.